src/detect-iprep.c

```diff
@@ -321,8 +322,8 @@ int DetectIPRepSetup (DetectEngineCtx *de_ctx, Signature *s, const char *rawstr)
     }
 
     if (value != NULL && strlen(value) > 0) {
-        int ival = atoi(value);
-        if (ival < 0 || ival > 127)
+        int8_t ival;
+        if (StringParseInt8(&ival, 10, 0, (const char *)value) < 0)
```

Comment:
Range check with 127 ?

Reply:
Should I use `uint8_t` instead and do the range check for 127 since it seems ival should always be positive?

Reply:
Yes, even better :-) and remove the later cast in `val = (uint8_t)ival;`